benchmarks/mempool.py

```diff
@@ -162,20 +178,76 @@ async def add_spend_bundles(spend_bundles: List[SpendBundle]) -> None:
                 tasks.append(asyncio.create_task(add_spend_bundles(spend_bundles[peer])))
             await asyncio.gather(*tasks)
             stop = monotonic()
-        print(f"add_spendbundle time: {stop - start:0.4f}s")
-        print(f"{(stop - start) / total_bundles * 1000:0.2f}ms per add_spendbundle() call")
+        print(f"  time: {stop - start:0.4f}s")
+        print(f"  per call: {(stop - start) / total_bundles * 1000:0.2f}ms")
 
-        with enable_profiler(True, f"create-{suffix}"):
+        print("\nProfiling add_spend_bundle() with replace-by-fee")
+        total_bundles = 0
+        tasks = []
+        with enable_profiler(True, f"replace-{suffix}"):
             start = monotonic()
-            for _ in range(2000):
-                mempool.create_bundle_from_mempool(bytes32(b"a" * 32))
+            for peer in range(NUM_PEERS):
+                total_bundles += len(replacement_spend_bundles[peer])
+                tasks.append(asyncio.create_task(add_spend_bundles(replacement_spend_bundles[peer])))
+            await asyncio.gather(*tasks)
             stop = monotonic()
-        print(f"create_bundle_from_mempool time: {stop - start:0.4f}s")
+        print(f"  time: {stop - start:0.4f}s")
+        print(f"  per call: {(stop - start) / total_bundles * 1000:0.2f}ms")
 
-    # TODO: add benchmark for new_peak()
+        print("\nProfiling create_bundle_from_mempool()")
+        with enable_profiler(True, f"create-{suffix}"):
+            start = monotonic()
+            for _ in range(500):
+                mempool.create_bundle_from_mempool(rec.header_hash)
+            stop = monotonic()
+        print(f"  time: {stop - start:0.4f}s")
+        print(f"  per call: {(stop - start) / 500 * 1000:0.2f}ms")
 
-    finally:
-        await db_wrapper.close()
+        print("\nProfiling new_peak() (optimized)")
+        blocks: List[Tuple[BlockRecord, NPCResult]] = []
+        for coin_id in all_coins.keys():
+            height = uint32(height + 1)
+            timestamp = uint64(timestamp + 19)
+            rec = fake_block_record(height, timestamp)
+            npc_result = NPCResult(
+                None,
+                SpendBundleConditions(
+                    [Spend(coin_id, bytes32(b" " * 32), None, 0, None, None, [], [], 0)], 0, 0, 0, None, None, [], 0
+                ),
+                uint64(1000000000),
+            )
+            blocks.append((rec, npc_result))
+
+        with enable_profiler(True, f"new-peak-{suffix}"):
+            start = monotonic()
+            for rec, npc_result in blocks:
+                await mempool.new_peak(rec, npc_result)
+            stop = monotonic()
+        print(f"  time: {stop - start:0.4f}s")
+        print(f"  per call: {(stop - start) / len(blocks) * 1000:0.2f}ms")
+
+        print("\nProfiling new_peak() (reorg)")
+        blocks = []
+        for coin_id in all_coins.keys():
+            height = uint32(height + 2)
+            timestamp = uint64(timestamp + 28)
```

Comment:
Is it intentional to advance by two blocks but only 28 seconds?

Reply:
yes, this is to trigger the slow, non-optimized, reorg case

Reply:
Right, but I meant the 28 seconds bit, shouldn't we opt for 38? (two blocks ~19 seconds per block)

Reply:
oh, it doesn't really matter. I don't think anything is looking at the timestamp in there, since we don't have any conditions